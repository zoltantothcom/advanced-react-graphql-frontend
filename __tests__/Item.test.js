import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ItemComponent from '../components/Item';

const fakeItem = {
  id: 'qw4hjt9q4owNF',
  title: 'Cool Item',
  price: 4000,
  description: 'This one is really cool!',
  image: 'cool.jpg',
  largeImage: 'large-cool.jpg',
};

describe('<Item />', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  xit('renders the pricetag and title properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);

    expect(
      wrapper
        .find('PriceTag')
        .children()
        .text(),
    ).toBe('$50');

    expect(
      wrapper
        .find('Title a')
        .first()
        .children()
        .text(),
    ).toBe(fakeItem.title);
  });

  xit('renders the image properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);

    const img = wrapper.find('img');

    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
  });

  xit('renders the buttons properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const buttonList = wrapper.find('.buttonList');

    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find('Link')).toHaveLength(1);
    expect(buttonList.find('AddToCart').exists()).toBe(true);
    expect(buttonList.find('DeleteItem').exists()).toBe(true);
  });
});
