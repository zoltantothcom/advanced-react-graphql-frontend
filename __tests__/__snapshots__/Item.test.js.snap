// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="Cool Item"
    src="cool.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "qw4hjt9q4owNF",
          },
        }
      }
    >
      <a>
        Cool Item
      </a>
    </Link>
    <PriceTag>
      $40
    </PriceTag>
    <p>
      This one is really cool!
    </p>
    <div
      className="buttonList"
    >
      <Link
        href={
          Object {
            "pathname": "update",
            "query": Object {
              "id": "qw4hjt9q4owNF",
            },
          }
        }
      >
        <a>
          Edit
        </a>
      </Link>
      <AddToCart
        id="qw4hjt9q4owNF"
      />
      <DeleteItem
        id="qw4hjt9q4owNF"
      >
        Remove Item
      </DeleteItem>
    </div>
  </Title>
</ItemStyles__Item>
`;
